### PR TITLE
Prevents everything except carbons and cats from resting to fix Ian paralysis.

### LINKED ISSUE #56034: Things phase through simplemobs if they are placed on tables using help intent and subsequently removed from said tables. Those simplemobs also become unable to move on their own and are pulled at about the same speed as prone/dead humans.
## Testmerges:
none (tested locally)

## Reproduction:
1. Find a simplemob that things shouldn't phase through (for example a cow or a corgi)
2. Pull it on help intent
3. Click a table
4. Wait for the progress bar to finish
5. Remove it from the table by destroying it or pulling it off the table
6. Walk through it (you phase through)
7. Throw something at it (it phases through)
8. Put your mind in the simplemob
9. Try to move (you can't, but you can rotate yourself)
10. Pull the mob (you will be slowed even if pulling the mob doesn't normally slow you down)

Possibly related to https://github.com/tgstation/tgstation/issues/44884? (edit: it's not)
https://github.com/tgstation/tgstation/issues/55133 is an issue about a specific situation in which this bug manifests

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
All living mobs (only living mobs have resting-related procs) now have a var that keeps track of their ability to rest. It it's false, mobs won't rest. Most living mobs shouldn't be able to rest, so it's only true for carbons and cats. Fixes #56034.

## Why It's Good For The Game
Fixes at least one bug.

## Changelog
:cl:
fix: Prevents forcing a lot of different kinds of mobs that shouldn't be able to rest to rest.
/:cl:

### CODE REVIEW

**Comment by coiax** [comment]
Can't pAIs also rest?

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -1,3 +1,6 @@
+/mob/living/carbon
+	can_rest = TRUE
```
Comment (Fikou) [inline code/modules/mob/living/carbon/carbon.dm:2]: WHY IS THIS HERE

Comment (interestingusernam3) [reply code/modules/mob/living/carbon/carbon.dm:2]: Why would it not be here?

**Comment by Qustinnus** [comment]
move the vars out of those files into their respective define files and turn it from a bool to either a bit flag or a trait

**Comment by interestingusernam3** [comment]
Into their respective what?

**Comment by interestingusernam3** [comment]
Making a new PR because I made a new branch for the flag version.

**Comment by Qustinnus** [comment]
why not just use this one

**Comment by interestingusernam3** [comment]
More efficient because I didn't have to undo everything before remaking the fix.
